### PR TITLE
NativoBidAdapter - Bid data mapping refactor and added QS params on request

### PR DESCRIPTION
**Pull request by jsfledd**
## Type of change

- [x] Refactoring (no functional changes, no api changes)

## Description of change
- Refactored the way we keep track of bids internally and map them to our system
- Added new query string parameters to the bid request

### CODE REVIEW

**Comment by ChrisHuie** [comment]
@jsfledd working on sorting out why tests are failing on the install step 🧐

**Review by mmoschovas — CHANGES_REQUESTED** [review]
@jsfledd can you drop the package.lock change here

**Comment by jsfledd** [comment]
@mmoschovas I have removed the package.lock from the PR

File: modules/nativoBidAdapter.js [hunk 1]
```diff
@@ -375,3 +466,14 @@ function getLargestSize(sizes, method = area) {
...
  */
 const area = (size) => size[0] * size[1]
+
+/**
+ * Save any filter data from winning bid requests for subsequent requests
+ * @param {Array} filter - The filter data bucket currently stored
+ * @param {Array} filterData - The filter data to add
+ */
+function appendFilterData(filter, filterData) {
+  if (filterData && Array.isArray(filterData) && filterData.length > 0) {
```
Comment (mmoschovas) [inline modules/nativoBidAdapter.js:476]: 0 is falsey so you can simply say 

`if (filterData && Array.isArray(filterData) && filterData.length) {`

**Comment by mmoschovas** [comment]
@jsfledd  Thank you for updating. One more super small change and this is good to go

**Review by mmoschovas — APPROVED** [review]

**Pull request merged**